src/sentry/incidents/logic.py

```diff
@@ -799,6 +799,30 @@ def subscribe_projects_to_alert_rule(alert_rule, projects):
     )
 
 
+def disable_alert_rule(alert_rule):
+    if alert_rule.status == AlertRuleStatus.DISABLED.value:
+        return
+    with transaction.atomic():
+        alert_rule.update(status=AlertRuleStatus.DISABLED.value)
+
+        # TODO: At the moment this also deletes the QuerySubscription from postgres AND the snuba subscription
+        # We might just want to delete the snuba subscription?
+        bulk_delete_snuba_subscriptions(alert_rule.snuba_query.subscriptions.all())
```

Comment:
We don't want to delete the Sentry subscriptions. At the moment they keep track of which projects the rule has been applied to. I think we should also be able to mark subscriptions as disabled on our end, which would just delete the snuba subscription.

Reply:
I added a flag to bulk_delete_snuba_subscriptions to disable the subscription instead of deleting it..and the task checks the status to determine whether to delete the subscription or not. What do you think?

Reply:
I think having a disabled status on the subscription makes sense.